### PR TITLE
Use tox, add dev tools

### PR DESCRIPTION
**Pull request by vladdu**
Tox is like 'make' but each set of commands can be run in their own virtual environment, with specific dependencies. https://tox.readthedocs.io/en/latest/

Run ./setup.sh first, then
- 'tox' will run tests
- 'tox-e clean,build' will build a distribution
- 'tox -e check' will run linters, but not modifiy files
- 'tox -e lint' will run linters and reformat files (using black)

I have not done anything to the release and publish process, there is a standard implementation.

I have fixed some warnings and errors, but there are more reported by pylint, that I'm not sure how to fix. If you think we should go with this, then I can look at them. I guess some are caused by "old" ways to do things.

Support to publishing docs to readthedocs.com can be added, if you think it would be better than using github pages. The yglu.io domain can be redirected there.

Even if you think this is overkill and we should not do it, please try to checkout the changes and see if you can run the tox commands. I'd like to know if it works in an unprepared environment. Thanks!

### CODE REVIEW

File: src/yglu/functions.py [hunk 1]
```diff
@@ -29,8 +29,8 @@ def import_function(filename):
             import_dirs.extend(import_path.split(os.pathsep))
 
         allow = False
-        for dir in import_dirs:
-            relative_path = os.path.relpath(filepath, start=dir)
+        for my_dir in import_dirs:
```
Comment (lbovet) [inline src/yglu/functions.py:32]: Could we give meaningful names like import_dir to renamed variables? my_ or a_ suffix are a bit noisy

**Comment by lbovet** [comment]
That's a big bunch of work, great! Many thanks and a warm welcome as project member!
I am a bit annoyed by the linting preventing to use variable names like 'input', 'stack' or 'dir'. Let's try to give useful alternative names instead of 'the_stack', 'my_dir' which really look like first programming lessons :) Or desactivate this rule...

**Comment by lbovet** [comment]
./setup.sh was successful, but then I have a missing dependency, I think:
```
23:30 $ tox -e clean,build
clean create: /home/shaman/work/yglu/.tox/clean
clean run-test-pre: PYTHONHASHSEED='1220048140'
clean run-test: commands[0] | python -c 'from shutil import rmtree; rmtree("build", True); rmtree("dist", True)'
build create: /home/shaman/work/yglu/.tox/build
build installdeps: build
build installed: build==0.3.1.post1,importlib-metadata==3.7.2,packaging==20.9,pep517==0.10.0,pyparsing==2.4.7,toml==0.10.2,typing-extensions==3.7.4.3,zipp==3.4.1
build run-test-pre: PYTHONHASHSEED='1220048140'
build run-test: commands[0] | python -m build .
The virtual environment was not created successfully because ensurepip is not
available.  On Debian/Ubuntu systems, you need to install the python3-venv
package using the following command.

    apt-get install python3-venv

You may need to use sudo with that command.  After installing the python3-venv
package, recreate your virtual environment.

Failing command: ['/tmp/build-env-gsshkm2u/bin/python', '-Im', 'ensurepip', '--upgrade', '--default-pip']

ERROR: InvocationError for command /home/shaman/work/yglu/.tox/build/bin/python -m build . (exited with code 1)
_________________________________________________________________ summary __________________________________________________________________
  clean: commands succeeded
ERROR:   build: commands failed
```

**Comment by vladdu** [comment]
I moved the pylint part to a different commit, so that we can go through the details in due time.

I'm not sure how to handle the python-venv thing, not everyone uses ubuntu so I don't think it should be in setup.sh.

**Comment by lbovet** [comment]
We can install venv via pip, no?

**Comment by vladdu** [comment]
I don't think so, at least it doesn't work for me (using Ubuntu too). Newer Python releases have venv in the standard libraries, older don't. Also the 'apt' packagers can decide to split distributions in multiple parts, for some reason.  

Having a complete Python environment installed should be a prerequisite for doing development, and I think we can't address all the possible strangeness of every Linux distribution (or OSX, or Windows...)

**Comment by lbovet** [comment]
Okay, so we just need to list it as a prerequisite.

**Comment by vladdu** [comment]
I checked around GitHub and no projects that I found were even mentioning this issue. Most say "install with 'pip install whatever'", some mention "pip install tox". Maybe it is not a big issue?

**Comment by lbovet** [comment]
Yes, I guess that any normal python dev has venv installed early in their
childhood. I may have a bit a lone wolf syndrome sometimes, not matching
with the herd :)

**Comment by lbovet** [comment]
I forgot a bit this issue. There is activity again on this project!

**Pull request merged**